framework/src/media/audio/audio_manager.c

```diff
@@ -951,8 +1087,29 @@ int start_audio_stream_out(void *data, unsigned int frames)
 		medvdbg("Resume the output audio card!!\n");
 	}
 
+	ret = get_supported_channel_num(OUTPUT, &device_channel_num);
+	if (ret != AUDIO_MANAGER_SUCCESS) {
+		meddbg("Fail to get channel number\n");
+		goto error_with_lock;
+	}
+	rechanneling_ratio = (float)device_channel_num / (float)card->resample.user_channel;
+
 	if (card->resample.necessary) {
-		resampled_frames = resample_stream_out(card, data, frames);
+		if (rechanneling_ratio > 1) {	// Mono -> Stereo
+			char *src = (char *)data;
+			char *dst = (char *)card->resample.rechanneling_buffer;
+			medvdbg("Start addr = %x\n", card->resample.rechanneling_buffer);
+			for (int i = 0; i < frames; i++) {
```

Comment:
it is different logic compare above and here.

Reply:
Aligned.